LongTasks/explainer.md

```diff
@@ -18,237 +18,215 @@ The Long Tasks API allows web developers to monitor long-running tasks on the ma
+In practice, we have observed long tasks lasting several minutes due to unexpected network delays or IndexedDB overhead.
 
-```js
-function longTaskMain() {
-  fibonacci(40);
-}
+# Proposal
 
-function longTaskWorker(url) {
-  const worker = new Worker(url);
-  worker.postMessage('startWorkerTask');
+We propose extending the Long Tasks API to support Web Workers, enabling developers to use the API within Web Workers to identify long tasks blocking the worker thread. To adapt the API for Web Workers, the `PerformanceLongTaskTiming` interface needs to be updated to account for long tasks in worker threads.
 
-  worker.onmessage = (event) => {
-    console.log('Worker task completed:', event.data);
-  };
-}
+The primary change would be within the `TaskAttributionTiming` interface, where certain properties should reflect the worker context instead of the main thread.
 
-longTaskMain();
-longTaskWorker("worker1.js");
-longTaskWorker("worker2.js");
-```
+## `TaskAttributionTiming` Interface Changes
 
-If you run the `PerformanceObserver` like this in the JavaScript console, you will get three performance entries as shown below:
+The table below highlights the differences in the properties of the `TaskAttributionTiming` interface for long tasks on Web Workers compared to the main thread:
 
+|**Property**|**Value(Main Thread)**|**Value(Web Worker)**|
+|---|---|---|
+|TaskAttributionTiming.duration|Always returns `0`|Always returns `0`|
+|TaskAttributionTiming.entryType|always returns `"taskattribution"`|Always returns `"taskattribution"`|
+|TaskAttributionTiming.name|always returns `"unknown"`| Return the task name given in postMessage API|
+|TaskAttributionTiming.startTime|always returns `0`|Always returns `0`|
+|TaskAttributionTiming.containerType|Returns the type of frame container(`iframe`, `embed`, or `object`). Defaults to `"window"` if no container is identified.| Returns the type of Web Worker (`"dedicated-worker"`, `"shared-worker"`, or `"service-worker"`).
+|TaskAttributionTiming.containerSrc|Returns the container's `src` attribute.|Returns the worker script's URL|
+|TaskAttributionTiming.containerId|Returns the container's `id` attribute.|Returns the worker's ID|
+|TaskAttributionTiming.containerName|Returns the container's `name` attribute.|Returns the `name` property of the worker|
+
+For workers: 
+- The `containerType` should return the type of Web Worker to clearly indicate the context.
+- The `containerSrc` provides the URL of the worker script, helping developers differentiate between multiple workers. 
+- If a name is specified for the Web Worker, the containerName property should return the assigned name.
+
+Example:
 ```js
-const observer = new PerformanceObserver((list) => {
-  console.log(list.getEntries());
-});
+const myWorker = new Worker("worker.js", { name: "Fibonacci" });
+```
+- When a name is given to postMessage in the following example, the name property of the `TaskAttributionTiming` inteface provide the task name.
```

Comment:
I don't see `name` as an [option in `postMessage()`](https://developer.mozilla.org/en-US/docs/Web/API/Worker/postMessage) currently. Are we also proposing the addition of this option?

Reply:
Yes, I need to add the name option in postMessage API.